### PR TITLE
style: --staged option

### PR DESCRIPTION
**Pull request by maxim-belkin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

`--staged` flag for `style` command to check only staged changes in files that have been staged for a commit.

### CODE REVIEW

**Comment by ilovezfs** [comment]
```
git diff --name-only --cached | xargs -n1 brew style
```

**Comment by maxim-belkin** [comment]
> ```
> git diff --name-only --cached | xargs -n1 brew style
> ```

That run `style` for all changes in the staged files.

**Comment by maxim-belkin** [comment]
OK, I clarified what the changes do: check only staged changes in staged files.

**Comment by ilovezfs** [comment]
I don't think we should be building git operations into `brew style`. I'd suggest putting this in a function in your bash profile or similar.

File: Library/Homebrew/cmd/style.rb [hunk 1]
```diff
@@ -35,6 +37,15 @@ def style
       ARGV.formulae.map(&:path)
     end
 
+    # `--staged`: detect staged files
+    if ARGV.include? "--staged"
+      staged = `git -C "#{HOMEBREW_REPOSITORY}" diff --cached --name-only`.split
```
Comment (DomT4) [inline Library/Homebrew/cmd/style.rb]: `Utils.popen_read` looks less grim. Just saying 🙈

Comment (maxim-belkin) [reply Library/Homebrew/cmd/style.rb]: Are you suggesting something like this:
```
staged = Utils.popen_read("git", "-C", HOMEBREW_REPOSITORY.to_s, "diff", "--cached", "--name-only").split
```

**Comment by ilovezfs** [comment]
Sorry @maxim-belkin going to pass on this PR.

File: Library/Homebrew/cmd/style.rb (hunk 1 above)
Comment (DomT4) [reply Library/Homebrew/cmd/style.rb]: Yeah, basically. I don't know if Homebrew has an official policy on this & literally any maintainer can overrule me but I never miss an opportunity to get anal about backticks.

**Comment by MikeMcQuaid** [comment]
Agreed here; I think things that can be achieved relatively simply on the shell are worth doing there unless there's a clear, wide demand for it.

**Pull request closed without merging**